control-plane-operator/controllers/hostedcontrolplane/multus_disable_test.go

```diff
@@ -0,0 +1,165 @@
+		},
+		{
+			name: "Explicit false should enable multus",
+			setupHCP: func() *hyperv1.HostedControlPlane {
+				return &hyperv1.HostedControlPlane{
+					Spec: hyperv1.HostedControlPlaneSpec{
+						OperatorConfiguration: &hyperv1.OperatorConfiguration{ClusterNetworkOperator: &hyperv1.ClusterNetworkOperatorSpec{DisableMultiNetwork: ptr.To(false)}},
+						Networking: hyperv1.ClusterNetworking{
+							NetworkType: hyperv1.OVNKubernetes,
+						},
+					},
+				}
+			},
+			expectedDisableMultiNetwork: ptr.To(false),
+			description:                 "When DisableMultiNetwork is explicitly set to false, multus should be enabled",
+		},
+		{
+			name: "Explicit true should disable multus",
+			setupHCP: func() *hyperv1.HostedControlPlane {
+				return &hyperv1.HostedControlPlane{
+					Spec: hyperv1.HostedControlPlaneSpec{
+						OperatorConfiguration: &hyperv1.OperatorConfiguration{ClusterNetworkOperator: &hyperv1.ClusterNetworkOperatorSpec{DisableMultiNetwork: ptr.To(true)}},
+						Networking: hyperv1.ClusterNetworking{
+							NetworkType: hyperv1.OVNKubernetes,
+						},
+					},
+				}
+			},
+			expectedDisableMultiNetwork: ptr.To(true),
+			description:                 "When DisableMultiNetwork is explicitly set to true, multus should be disabled",
+		},
+		{
+			name: "DisableMultiNetwork works with empty networking config",
+			setupHCP: func() *hyperv1.HostedControlPlane {
+				return &hyperv1.HostedControlPlane{
+					Spec: hyperv1.HostedControlPlaneSpec{
+						OperatorConfiguration: &hyperv1.OperatorConfiguration{ClusterNetworkOperator: &hyperv1.ClusterNetworkOperatorSpec{DisableMultiNetwork: ptr.To(true)}},
+						Networking:            hyperv1.ClusterNetworking{
+							// Minimal networking config
+						},
+					},
+				}
+			},
+			expectedDisableMultiNetwork: ptr.To(true),
+			description:                 "DisableMultiNetwork should work even with minimal networking configuration",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			hcp := test.setupHCP()
+
+			// Test that the field value matches expectations
+			if test.expectedDisableMultiNetwork == nil {
+				assert.Nil(t, hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork, test.description)
+			} else {
+				assert.NotNil(t, hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork, test.description)
+				assert.Equal(t, *test.expectedDisableMultiNetwork, *hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork, test.description)
+			}
+
+			// Test that the reconciler logic would make the correct decision
+			// When DisableMultiNetwork is nil, it defaults to false (multus enabled)
+			disableMultiNetwork := false
+			if hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork != nil {
+				disableMultiNetwork = *hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork
+			}
+			shouldProcessMultus := !disableMultiNetwork
```

Comment:
It's not clear to me what main controller logic this is testing. If this intent is to imitate the logic in lines 2309-2316, then that logic should be extracted in a separate function and the function called from the test. Otherwise, it's easy to update that logic in the future without updating the logic here in the test.